thingsboard_gateway/connectors/mqtt/mqtt_connector.py

```diff
@@ -269,6 +270,42 @@ def get_type(self):
     def get_ttl_for_duplicates(self, device_name):
         return self.__send_data_only_on_change_ttl
 
+    @staticmethod
+    def __add_ts_to_test_message(msg, ts_name):
+        msg = simplejson.loads(msg.decode('utf-8').replace("'", '"'))
+        msg[ts_name] = int(time() * 1000)
+        return simplejson.dumps(msg).encode('utf-8')
+
+    def check_message_latency(self):
+        try:
+            test_converter_config = {
+                'converter': {
+                    'deviceInfo': {
+                        'deviceNameExpressionSource': 'constant',
+                        'deviceNameExpression': 'currentThingsBoardGateway',
+                        'deviceProfileExpressionSource': 'constant',
+                        'deviceProfileExpression': 'default',
+                    },
+                    'timeseries': [
+                        {'key': 'beforeConversionTs', 'type': 'int', 'value': '${beforeConversionTs}'},
+                        {'key': 'receivedTs', 'type': 'int', 'value': '${receivedTs}'},
+                        {'key': 'connectorName', 'type': 'string', 'value': '${connectorName}'},
+                        {'key': 'isTestLatencyMessageType', 'type': 'bool', 'value': '${isTestLatencyMessageType}'}
+                    ]
+                }
+            }
+            converter = JsonMqttUplinkConverter(test_converter_config, self.__log)
+            self.__mapping_sub_topics['test/latency'] = [converter]
```

Comment:
Please move topic to connector level and generate it randomly to avoid intersection with topics that can be used by users.